src/code/graph.c

```diff
@@ -113,27 +113,28 @@ void Graph_InitTHGA(GraphicsContext* gfxCtx) {
     gfxCtx->unk_014 = 0;
 }
 
+// Used below in `Graph_GetNextGameState`
+#define DEFINE_GAMESTATE_INTERNAL(typeName) GAMESTATE_INDEX_##typeName,
+#define DEFINE_GAMESTATE(typeName, name) DEFINE_GAMESTATE_INTERNAL(typeName)
+typedef enum {
+#include "tables/gamestate_table.h"
+    GAMESTATE_INDEX_MAX
+} GameStateTableIndex;
+#undef DEFINE_GAMESTATE
+#undef DEFINE_GAMESTATE_INTERNAL
```

Comment:
I would prefer this to be in a header instead

Reply:
I'm not sure there's really any merit to being in a header besides avoiding potential type clashes. The enum values are not "nice", e.g. the PlayState enum value is `GAMESTATE_ENUM_Play`, it serves no purpose other than to be used in `Graph_GetNextGameState`

Reply:
While I know it would only get this one use, it still seems nice to have a global enum (pretty sure this would be nice if you wanted to add an `Id` field to the `GameState`). Personally I think it would be nice to add a second/third argument for an enum value that is also defined in the table (like actors), and use that instead of the concatenation. Could also add one for the init function name, but that is even more kind of whatever.